src/test/java/api/CirculationRulesAPITests.java

```diff
@@ -161,14 +163,97 @@ public void cannotUpdateCirculationRulesWithLostItemPolicyId() {
       overdueFinePoliciesFixture.facultyStandard().getId().toString(),
       UUID.randomUUID().toString());
 
-    Response response = circulationRulesFixture
-      .attemptUpdateCirculationRules(rule, "i");
+    Response response = circulationRulesFixture.attemptUpdateCirculationRules(rule);
 
     assertThat(response.getStatusCode(), is(422));
     assertThat(response.getJson().getString("message"),
       is("The policy i does not exist"));
   }
 
+  @Test
+  public void canUpdateCirculationRulesWithTwentyExistingLoanPolicies() {
+
+    Set<UUID> loanPolicyIds = getSetOfPolicyIds(20);
+    addPoliciesToLoanPolicyFixture(loanPolicyIds);
+
+    loanPolicyIds.forEach(loanPolicyId -> {
+      String rule = circulationRulesFixture.soleFallbackPolicyRule(
+        loanPolicyId.toString(), requestPoliciesFixture.allowAllRequestPolicy().getId().toString(),
+        noticePoliciesFixture.activeNotice().getId().toString(),
+        overdueFinePoliciesFixture.facultyStandard().getId().toString(),
+        lostItemFeePoliciesFixture.facultyStandard().getId().toString());
+      circulationRulesFixture.updateCirculationRules(rule);
+    });
+  }
+
+  @Test
+  public void canUpdateCirculationRulesWithTwentyExistingNoticePolicies() {
+
+    Set<UUID> noticePolicyIds = getSetOfPolicyIds(20);
+    addPoliciesToNoticePolicyFixture(noticePolicyIds);
+
+    noticePolicyIds.forEach(noticePolicyId -> {
+      String rule = circulationRulesFixture.soleFallbackPolicyRule(
+        loanPoliciesFixture.canCirculateFixed().getId().toString(),
+        requestPoliciesFixture.allowAllRequestPolicy().getId().toString(),
+        noticePolicyId.toString(),
+        overdueFinePoliciesFixture.facultyStandard().getId().toString(),
+        lostItemFeePoliciesFixture.facultyStandard().getId().toString());
+      circulationRulesFixture.updateCirculationRules(rule);
+    });
+  }
+
+  @Test
+  public void canUpdateCirculationRulesWithTwentyExistingRequestPolicies() {
+
+    Set<UUID> requestPolicyIds = getSetOfPolicyIds(20);
+    requestPolicyIds.forEach(requestPoliciesFixture::allowAllRequestPolicy);
+
+    requestPolicyIds.forEach(requestPolicyId -> {
+      String rule = circulationRulesFixture.soleFallbackPolicyRule(
+        loanPoliciesFixture.canCirculateFixed().getId().toString(),
+        requestPoliciesFixture.allowAllRequestPolicy(requestPolicyId).getId().toString(),
```

Comment:
Can you extract this line?

    UUID allowAllRequestPolicyId = requestPoliciesFixture.allowAllRequestPolicy(requestPolicyId).getId();
    String rule = ...

Reply:
updated